tests/hkdf.cc

```diff
@@ -0,0 +1,239 @@
+/*
+ *    Copyright 2022 Cryspen Sarl
+ *
+ *    Licensed under the Apache License, Version 2.0 or MIT.
+ *    - http://www.apache.org/licenses/LICENSE-2.0
+ *    - http://opensource.org/licenses/MIT
+ */
+
+#include <fstream>
+
+#include <gtest/gtest.h>
+#include <nlohmann/json.hpp>
+
+#include "Hacl_HKDF.h"
+#include "Hacl_Spec.h"
+#include "util.h"
+
+using json = nlohmann::json;
+
+typedef struct
+{
+  std::string comment;
+  // A tcId of 0 indicates "not Wycheproof".
+  uint32_t tcId;
+  std::string hash;
+  // 1) Input Keying Material
+  bytes IKM;
+  bytes salt;
+  bytes info;
+  // Length of OKM
+  uint32_t L;
+  // 2) A pseudorandom key (of HashLen octets)
+  // Only available when "not Wycheproof".
+  bytes PRK;
+  // 3) Output Keying Material
+  bytes OKM;
+  bool valid;
+} TestCase;
+
+class HKDFSuite : public ::testing::TestWithParam<TestCase>
+{};
+
+TEST_P(HKDFSuite, TestCase)
+{
+  auto test = GetParam();
+  std::cout << "# " << test.comment << std::endl;
+
+  // TODO: How does error handling in Hacl_HKDF_* work?
+  if (!test.valid) {
+    return;
+  }
+
+  if (test.hash == "SHA-1") {
```

Comment:
If SHA-1 is not supported, we don't need the json test vector for it.

Reply:
The `tests/hkdf/rfc5869.json` contains all test vectors from RFC 5869. JSON does not allow comments and I think we should at least not "silently" remove the SHA-1 test vectors from the JSON file without documenting it somewhere. I could document it in the tests/README.md or leave it as it is.

Reply:
I silenced the skip part.

Reply:
Right, but we don't need the sha1 json file then 🙂